File: epi_judge_python/int_square_root.py
from test_framework import generic_test


def square_root(k):
    l, r = 0, 2 ** 16

    while l <= r:
        m = (l + r)//2
        if m ** 2 <= k and (m + 1) ** 2 > k:
            return m
        elif m ** 2 < k:
            l = m + 1
        else:
            r = m - 1
    return 0


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main("int_square_root.py",
                                       "int_square_root.tsv", square_root))

File: epi_judge_java/epi/IntSquareRoot.java
package epi;
import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;
public class IntSquareRoot {
  @EpiTest(testDataFile = "int_square_root.tsv")

  public static int squareRoot(int k) {
    if (k <= 1) return k;

    long l = 1, r = k;
    long guess = 1;
    while (l < r){

      guess = (l + r)/2;
      if (guess * guess <= k && Math.pow(guess+1,2) > k){
        return (int) guess;
      } else if (guess * guess > k){
        r = guess - 1;
      } else {
        l = guess + 1;
      }
    }
    return (int) l;
  }

  public static void main(String[] args) {
    System.exit(
        GenericTest
            .runFromAnnotations(args, "IntSquareRoot.java",
                                new Object() {}.getClass().getEnclosingClass())
            .ordinal());
  }
}
